test/atom_finder/classifier_test.clj

```diff
@@ -151,3 +151,7 @@
 (deftest test-assignment-as-value-atom?
   (testing "assignment-as-value-atom? finds all atoms in snippet study code"
     (test-atom-lines "assignment-as-value.c" "<true>" (default-finder assignment-as-value-atom?))))
+
+(deftest test-implicit-predicate-atom?
+  (testing "implicit-predicate-atom? finds all atoms in sample code")
+    (test-atom-lines "implicit-predicate.c" "<true>" (default-finder implicit-predicate-atom?)))
```

Comment:
Can you add individual tests to test at the node (not line) level? See for example: https://github.com/dgopstein/atom-finder/blob/master/test/atom_finder/classifier_test.clj#L14

Then add a few tests. For example I'm curious what your code says for `!1` and `!1 ? 2 : 3`. In theory I think the first one should be an atom, and the second one shouldn't.